overlord/devicestate/firstboot.go

```diff
@@ -142,12 +139,17 @@ func populateStateFromSeedImpl(st *state.State, opts *populateStateFromSeedOptio
 		return nil, err
 	}
 
+	model := deviceSeed.Model()
+
 	essentialSeedSnaps := deviceSeed.EssentialSnaps()
 	seedSnaps, err := deviceSeed.ModeSnaps(mode)
 	if err != nil {
 		return nil, err
 	}
 
+	// optimistically forget the deviceSeed here
+	unloadDeviceSeed(st)
```

Comment:
I wonder if it would make sense to make "unloadDeviceSeed" a defer and move it closer to importAssertionsFromSeed()? This would make it more "local" and one needs to keep less state/details in mind when reading this code. Or is there a deeper reason that it's this particular point or a downside of doing it as a defer slight further up? Fwiw, I was thinking something like:
```
diff --git a/overlord/devicestate/firstboot.go b/overlord/devicestate/firstboot.go
index 9113a888e1..0be0e97130 100644
--- a/overlord/devicestate/firstboot.go
+++ b/overlord/devicestate/firstboot.go
@@ -120,10 +120,11 @@ func populateStateFromSeedImpl(st *state.State, opts *populateStateFromSeedOptio
        if err != nil && err != errNothingToDo {
                return nil, err
        }
-
        if err == errNothingToDo {
                return trivialSeeding(st), nil
        }
+       // forget the deviceSeed again
+       defer unloadDeviceSeed(st)
 
        timings.Run(tm, "load-verified-snap-metadata", "load verified snap metadata from seed", func(nested timings.Measurer) {
                err = deviceSeed.LoadMeta(nested)
@@ -147,9 +148,6 @@ func populateStateFromSeedImpl(st *state.State, opts *populateStateFromSeedOptio
                return nil, err
        }
 
-       // optimistically forget the deviceSeed here
-       unloadDeviceSeed(st)
-
        tsAll := []*state.TaskSet{}
        configTss := []*state.TaskSet{}
```

Reply:
I did it this way considering we don't want to forget it if we error, but maybe that is not a valid concern or maybe it is good to forget if we error

Reply:
I didn't do anything about this, I need to do a follow up to change this